### DISCUSSION TITLE
Updating a Json field in postgresql does not remove properties from the json in the DB

### DISCUSSION THREAD

**Discussion by kareldonk** [mikro-orm/mikro-orm#5382] [Q&A]
### Describe the bug

Let's say that I have a `document` table in PostgreSQL with a column named `metadata` of type `jsonb`. 
And I have a `Document` entity in mikro-orm with a `metadata` property of type `any` like this:

```
class Document {
    @Property({ nullable: true, type: 'jsonb' })
    public metadata?: any;
}
```

I do an update on the table for a record, and send the following json in the metadata:

```
{
      "propa": "test",
      "propb": "test",
      "propc": "test",
}
```

After that I do another update:

```
{
      "propa": "test",
      "propd": "test",
}
```

The result in the DB is:

```
{
      "propa": "test",
      "propb": "test",
      "propc": "test",
      "propd": "test",
}
```

It appears the properties get merged instead of replaced. Is there a way to influence this?

### Reproduction

see description.

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6

### Node.js version

18+

### Operating system

windows

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Please provide reproduction as code, its not clear what you are doing.

edit: my guess would be you calling `assign()` with `mergeObjectProperties: true`?

**Comment by kareldonk** [comment]
Thanks for the quick response.

Since this is production code I cannot provide a full working example, but I will try to explain here.
What I do is basically this:

1. Get the document from the DB with this function:

```
let document = this.documentRepository.findOne({ id: 'ID_HERE' });
```

2. Then further down the code I update the document with this:

```
const entityData: EntityData<Document> = {
...updateDocumentDto
};
await this.repository.upsert(entityData);
```

The `updateDocumentDto` is just an object with the same properties as the `Document`.

When I enable the `DATABASE_DEBUG` environment variable, I can see that the SQL that gets sent to the DB includes the merged properties in the `metadata` column. 

Now here's the interesting thing, if I set the `metadata` column directly on the document object I initially retrieved, then it does overwrite the value in the DB instead of merging:

```
const entityData: EntityData<Document> = {
...updateDocumentDto
};
document.metadata = updateDocumentDto.metadata; // This line added
await this.repository.upsert(entityData);
```

Is it perhaps possible that mikro-orm is comparing the metadata objects in memory for the document entitity that was initially retrieved (and probably in cache) and merging them before sending the SQL update query?
